### ISSUE TITLE
Update to Among Us 2023.7.12

### ISSUE THREAD

**Issue by Alexejhero** [SubmergedAmongUs/Submerged#56]
The Submerged update for v2023.6.13 is very close, we are going to do some beta tests and then we will publish the new version and also open-source the mod.

<details>
<summary>Previous checklist</summary>

Required changes:
- [x] Adjust no shadow objects
- [x] Adjust credits menu
- [x] Cannot start the game as the following players do not have Submerged
- [x] Fix engine move vent animation
- [x] Fix skin animation not working in Oxygen death cutscene
- [x] Fix skin animation not working when spawning in
- [x] Make sure round-robin impostor works
- [x] Fix steady heartbeat task
- [x] Upper decontam not showing in locations
- [x] Fix TranslationController.GetString error
- [x] Fix compile errors
- [x] Adjust create game / find game UI
- [x] Create HnS flow for Submerged
  - [x] Spawn in minigame
    - [x] Allow only impostor to choose spawn in
    - [x] Display spawn in minigame before the crewmate hide time starts
    - [x] Spawn all other players at the impostor's spawn location
    - [x] Fix fade overlay size
  - [x] Consoles
    - [x] Prevent central and admin vents from being used
    - [x] Prevent elevators from being used
    - [x] Prevent cameras from being sabotaged
  - [x] Floors
    - [x] Prevent map scrolling
    - [x] Close elevators at start
    - [x] Prevent ghosts from switching floors
  - [x] Tasks
    - [x] Prevent other floor tasks from appearing
    - [x] Adjust tasks to only have consoles on correct floor during HnS
      - [x] Plug Leaks
      - [x] Fix Wiring
      - [x] Mop Puddles
      - [x] Upload Data
    - [x] Balance tasks for HnS
      - [x] Microwave Lunch
      - [x] Spot Whale Shark
- [x] Fix not being able to login if Submerged is installed
- [x] Fix haunting not teleporting ghost to target player's floor
- [x] Fix HnS exception caused by missing emergency button
- [x] Fix lower central vent can use patch
- [x] Client DC because Error

</details>

**Comment by AlexejheroYTB** [comment]
I never got around to updating to 2022.12.14 because the I was busy recently. I will work on updating Submerged to 2023.2.9 instead. (#61)

**Comment by AlexejheroYTB** [comment]
All of the required features for updating to the latest version have been implemented. All that remains now is to fix the various bugs that were spotted.

**Comment by Alexejhero** [comment]
Update: We are very close to finishing the update for version 2023.6.13, once Reactor and BepInEx update we will update as well and also open-source the mod

**Comment by Alexejhero** [comment]
The mod will be licensed under [CC BY-NC-ND 4.0](https://creativecommons.org/licenses/by-nc-nd/4.0/)

**Comment by Tommy-XL** [comment]
Oh, thanks👍

**Comment by PHGaming121** [comment]
Reactor updated, now can u guys update the mod?

**Comment by Alexejhero** [comment]
yep we plan on beta testing today or tomorrow and then releasing sometime during the next week

**Comment by PHGaming121** [comment]
ok

**Comment by zozogame1** [comment]
YEAAAH 😄

**Comment by Wael1231** [comment]
very happy for submerged  if you can release new version tomorrow

**Comment by Wael1231** [comment]
And i also make an website for my best map Submerged https://submerged-website.vercel.app anyone can vist it

**Comment by AnthoYt** [comment]
A specific date for next week?

**Comment by AnthoYt** [comment]
A specific date for next week?

**Comment by Alexejhero** [comment]
no we don't know yet

**Comment by AnthoYt** [comment]
Ok, we can't wait for us and our team to do lots of live on it, but is it still scheduled for next week or late?

**Comment by Alexejhero** [comment]
Our current plans are to release a beta version for testing sometime next week

**Comment by AnthoYt** [comment]
Ok, I trust you, all my team wishes you good luck

**Comment by AnthoYt** [comment]
Is possible to have the mod to test it and tell you the problems or bugs ??

**Comment by zozogame1** [comment]
Oh yeah plz !

**Comment by AnthoYt** [comment]
J'ai demandé en premier...

**Comment by zozogame1** [comment]
Oui bon bah, si ils disent oui alors je pense qu'il y aura des versions beta pour tout le monde

**Comment by AnthoYt** [comment]
Oui, je rigole Bg j'attends une nouvelle version pour faire best of avec mes potes.. !

**Comment by zozogame1** [comment]
French : (for AnthoYT)
Par contre, j'ai toujours pas compris comment on fait pour jouer en ligne (j'avais essayé sur une ancienne version et sa ne marchait pas)

English: (for the developers)
However, I still haven't figured out how to play online (I tried it on an older version and it didn't work).

**Comment by Alexejhero** [comment]
Just wait for the new update please

**Comment by AnthoYt** [comment]
@zozogame1 pour jouer en ligne avec le mod ou sans le mod ?

**Comment by zozogame1** [comment]
> Just wait for the new update please

Ok

**Comment by zozogame1** [comment]
> @zozogame1 pour jouer en ligne avec le mod ou sans le mod ?

Avec

**Comment by AnthoYt** [comment]
Bas normale car avec une ancienne version la map n'est pas dispo mais je comprends pas pourquoi tu peux pas jouer en linge.. j'aimerais bien te demander ton discord pour t'aider mais je sais pas si t'es français ou anglais...

**Comment by zozogame1** [comment]
> Bas normale car avec une ancienne version la map n'est pas dispo mais je comprends pas pourquoi tu peux pas jouer en linge.. j'aimerais bien te demander ton discord pour t'aider mais je sais pas si t'es français ou anglais...

Français mais sa sert à rien comme j'ai mis à jour le jeux donc la map ne marche plus. Je vais juste attendre que la map soit mise à jour. Merci quand même

**Comment by AnthoYt** [comment]
Ok, bas si t'as encore le problème avec la mise à jour de la map, ping moi sur ce site et je viendrais t'aider avec plaisir.

**Comment by zozogame1** [comment]
Ok

**Comment by AnthoYt** [comment]
@natsu-k Désolé t'as raison je me suis laissé emporté

**Comment by zozogame1** [comment]
> @zozogame1 @AnthoYt
> 
> Please take your personal conversation elsewhere, this is not discord.

Sorry

**Comment by Alexejhero** [comment]
A beta version has been released: https://github.com/SubmergedAmongUs/Submerged/releases/tag/v2023.6.20-beta.1

**Comment by GeekCornerGH** [comment]
> The mod will be licensed under [CC BY-NC-ND 4.0](https://creativecommons.org/licenses/by-nc-nd/4.0/), but we will allow snippets of submerged code to be taken and used by other projects.
> 
> The purpose of this license is to prevent someone from forking the project, making one small change and then releasing it as "submerged+" like what has happened with many many other role mods in the past

Could you explain briefly why wasn't the mod open source? Polus.gg had a part of legit, but since the server is gone, I can't really think of any reason why this wasn't done before

**Comment by AnthoYt** [comment]
Merci pour la version

**Comment by GeekCornerGH** [comment]
> Hi. You are mistaking polusgg with submerged. This is a common error people make.
> 
> Submerged is a custom map made for Among Us. It was a paid project, privately funded and created for a group of content creators. The map never needed a public release, but was generously provided to the public by the owners through an agreement with polusgg which was a separate and unrelated product and company to submerged.
> 
> After polusgg went down we (the submerged team) investigated options to provide it to the public, which resulted in this GitHub distribution. However, Submerged has and continues to be privately owned.
> 
> Realistically, this privately funded and owned project concluded a long time ago and the public release has been kept updated through the generosity of the remaining active devs. However, the developers cannot continue to freely update the project consistently as it is very time consuming and unsustainable in the long term. This is part of the reasoning behind open sourcing the project.
> 
> Why wasn't it open sourced earlier? Because we were still exploring our available options to provide it to the public in the long term, and it's continued to be provided as a distributable where time has been available to update it - so it was never necessary.

Thanks for the explaination, I didn't taken polusgg and submerged, just misexplained myself. But thanks for the explaination, definitly makes sense.

**Comment by tartsharpcobra** [comment]
this should be closed

**Comment by Alexejhero** [comment]
Il2CppInterop pull request has been merged, we're waiting for a BepInEx update and then we will release.

**Comment by AnthoYt** [comment]
@JeanAU il y a des version bêtas qui sont disponible

**Comment by JeanAU** [comment]
> @JeanAU il y a des version bêtas qui sont disponible

yes i know

**Comment by Alexejhero** [comment]
An update for Submerged is ready. BepInEx has also been updated. We are waiting for Reactor to update to latest and then we will release.

As for open-sourcing, we are required to postpone that for a bit as there's still some problems we need to address first before we can do that.

**Comment by JeanAU** [comment]
> An update for Submerged is ready. BepInEx has also been updated. We are waiting for Reactor to update to latest and then we will release.
> 
> As for open-sourcing, we are required to postpone that for a bit as there's still some problems we need to address first before we can do that.

Thanks i'am waiting very excited

**Comment by JeanAU** [comment]
@Alexejhero  BepInEx and Reactor has release new version so when submerged get updated ?

**Comment by Alexejhero** [comment]
Done

**Comment by zozogame1** [comment]
Yeah !
